core/src/main/java/org/testcontainers/containers/wait/strategy/HttpWaitStrategy.java

```diff
@@ -150,8 +150,17 @@ protected void waitUntilReady() {
         if (null == livenessCheckPort || -1 == livenessCheckPort) {
             return;
         }
-        final String uri = buildLivenessUri(livenessCheckPort).toString();
-        log.info("{}: Waiting for {} seconds for URL: {}", containerName, startupTimeout.getSeconds(), uri);
+        final URI rawUri = buildLivenessUri(livenessCheckPort);
+        // Un-map the port for logging
+        int originalPort = -1;
+        for (Integer exposedPort : waitStrategyTarget.getExposedPorts()) {
+            if (rawUri.getPort() == waitStrategyTarget.getMappedPort(exposedPort)) {
```

Comment:
`getMappedPort` may throw, please make sure you handle it too

Reply:
I'm struggling to think of a scenario where this could throw and we'd want to continue...

* If the container isn't started, waiting for the port should not succeed and should fail fast
* If the port that was configured to be exposed cannot be mapped, something has gone seriously wrong with Docker

Isn't it therefore correct to just let any failure bubble up immediately? Or have I missed a scenario?

Reply:
https://github.com/testcontainers/testcontainers-java/blob/780b8e33811f046e64c7cdf6a25b49fb0fe79ce8/core/src/main/java/org/testcontainers/containers/ContainerState.java#L141 and https://github.com/testcontainers/testcontainers-java/blob/780b8e33811f046e64c7cdf6a25b49fb0fe79ce8/core/src/main/java/org/testcontainers/containers/ContainerState.java#L152

Reply:
A custom implementation may also be provided. So I would make a best-effort attempt at obtaining the port, but not fail the strategy if i throws

Reply:
Yeah, those are the two scenarios I was referencing. When I looked, I couldn't seen anything overridable (i.e. this only calls private methods), but maybe I missed something.

I'll have another look.

Reply:
I'm still not sure we strictly need it, but I've added a try/catch anyway.